nixos/modules/system/boot/luksroot.nix

```diff
@@ -294,13 +294,11 @@ let
 
         new_response="$(ykchalresp -${toString yubikey.slot} -x $new_challenge 2>/dev/null)"
 
-        if [ ! -z "$k_user" ]; then
-            new_k_luks="$(echo -n $k_user | pbkdf2-sha512 ${toString yubikey.keyLength} $new_iterations $new_response | rbtohex)"
-        else
-            new_k_luks="$(echo | pbkdf2-sha512 ${toString yubikey.keyLength} $new_iterations $new_response | rbtohex)"
-        fi
-
-        echo -n "$new_k_luks" | hextorb > /crypt-ramfs/new_key
+        if [ -n "$k_user" ]; then
+	          echo -n "$k_user"
+	      else
+            echo
+	      fi | pbkdf2-sha512 ${toString yubikey.keyLength} $new_iterations $new_response > /crypt-ramfs/new_key
```

Comment:
Lines 298, 299, 301 contain tabs. These should be replaced with spaces.

Reply:
Fixed.